src/course-unit/move-modal/hooks.tsx

```diff
@@ -0,0 +1,234 @@
+
+import { RequestStatus } from '../../data/constants';
+import { useEventListener } from '../../generic/hooks';
+import { getCourseOutlineInfo, getCourseOutlineInfoLoadingStatus } from '../data/selectors';
+import { getCourseOutlineInfoQuery, patchUnitItemQuery } from '../data/thunk';
+import { useIframe } from '../context/hooks';
+import { messageTypes } from '../constants';
+import {
+  CATEGORIES_KEYS, CATEGORIES_TEXT, CATEGORY_RELATION_MAP, MOVE_DIRECTIONS,
+} from './constants';
+import {
+  findParentIds, getBreadcrumbs, getXBlockType, isValidCategory,
+} from './utils';
+import {
+  IState, IUseMoveModalParams, IUseMoveModalReturn, IXBlockInfo,
+} from './interfaces';
+
+// eslint-disable-next-line import/prefer-default-export
+export const useMoveModal = ({
+  isOpenModal, closeModal, openModal, courseId,
+}: IUseMoveModalParams): IUseMoveModalReturn => {
+  const { blockId } = useParams<{ blockId: string }>();
+  const intl = useIntl();
+  const dispatch = useDispatch();
+  const { sendMessageToIframe } = useIframe();
+  const courseOutlineInfo = useSelector(getCourseOutlineInfo);
+  const loadingStatus = useSelector(getCourseOutlineInfoLoadingStatus);
+
+  const initialValues = useMemo<IState>(() => ({
+    childrenInfo: { children: courseOutlineInfo.child_info?.children ?? [], category: CATEGORIES_KEYS.section },
+    parentInfo: { parent: courseOutlineInfo, category: CATEGORIES_KEYS.course },
+    isValidMove: false,
+    sourceXBlockInfo: { current: {} as IXBlockInfo, parent: {} as IXBlockInfo },
+    visitedAncestors: [courseOutlineInfo],
+  }), [courseOutlineInfo]);
+
+  const [state, setState] = useState<IState>(initialValues);
+
+  const isExtraSmall = useMediaQuery({ maxWidth: breakpoints.extraSmall.maxWidth });
+
+  const currentXBlockParentIds = useMemo(
+    () => findParentIds(courseOutlineInfo, state.sourceXBlockInfo.current.id as string),
+    [courseOutlineInfo, state.sourceXBlockInfo.current.id],
+  );
+
+  const receiveMessage = useCallback(({ data }: { data: any }) => {
+    const { payload, type } = data;
+
+    if (type === messageTypes.showMoveXBlockModal) {
+      setState((prevState) => ({
+        ...prevState,
+        sourceXBlockInfo: {
+          current: payload.sourceXBlockInfo,
+          parent: payload.sourceParentXBlockInfo,
+        },
+      }));
+      openModal();
+    }
+  }, [openModal]);
+
+  useEventListener('message', receiveMessage);
+
+  const updateParentItemsData = useCallback((direction?: string, newParentIndex?: string) => {
+    setState((prevState: IState) => {
+      if (direction === undefined) {
+        return {
+          ...prevState,
+          parentInfo: {
+            parent: initialValues.parentInfo.parent,
+            category: initialValues.parentInfo.category,
+          },
+          visitedAncestors: [initialValues.parentInfo.parent],
+        };
+      }
+
+      if (
+        direction === MOVE_DIRECTIONS.forward && newParentIndex !== undefined
```

Comment:
Let's leave it without changes because we have dependency inside conditions.
Also storing child code inside particular condition looks more clear and simple for read.